core/templates/dev/head/domain/exploration/HintObjectFactory.js

```diff
@@ -0,0 +1,40 @@
+// Copyright 2015 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of Hint
+ * domain objects.
+ */
+
+oppia.factory('HintObjectFactory', function() {
+  var Hint = function(hintText) {
+    this.hint_text = hintText;
+  };
+
+  Hint.prototype.toBackendDict = function() {
+    return {
+      hint_text: this.hint_text
+    };
+  };
+
+  Hint.createFromBackendDict = function(hintBackendDict) {
+    return new Hint(hintBackendDict.hint_text);
+  };
+
+  Hint.createDefault = function(hintText) {
```

Comment:
I am mostly confused about the overlap between this PR and the other PR. I'm fine with whatever you do as long as you end up with a clean PR that can be given a once-over before merge into develop.

One thought, btw. Do you want to disable the hint editor in the frontend until you have the whole pipeline built out? Otherwise if someone is using the fallbacks (which are auto-migrated to hints), and then edits the hints directly, you're going to end up with a bit of a mess.